src/rules/tryceratops/rules/verbose_raise.rs

```diff
@@ -0,0 +1,47 @@
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{Excepthandler, ExcepthandlerKind, ExprKind, Located, StmtKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct VerboseRaise;
+);
+impl Violation for VerboseRaise {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use 'raise' without specifying exception name")
+    }
+}
+
+/// TRY201
+pub fn verbose_raise(checker: &mut Checker, handlers: &[Excepthandler]) {
+    for handler in handlers {
+        let ExcepthandlerKind::ExceptHandler { name, body, .. } = &handler.node;
+        // if the handler assigned a name to the exception
+        if let Some(name) = name {
+            // look for `raise` statements in the body
+            for stmt in body {
```

Comment:
I think we need to walk the tree recursively here. As-is, we'll miss this case:

```py
def bad():
    try:
        process()
    except MyException as e:
        logger.exception("process failed")
        if True:
            raise e
```

Since `stmt` here would be the `StmtKind::If`. We have to go one level further.

Here's how I'd recommend doing it...

If you look in `src/rules/flake8_annotations/rules.rs`, we have a struct called `ReturnStatementVisitor`. It just iterates over a bunch of statements recursively and collects all the `StmtKind::Return` statements. Here, we can add a `RaiseStatementVisitor` that does the same thing. Then you can iterate over the raises, and keep the logic below.

Reply:
Cool!

any chance that we are missing `ClassDef` there?
https://github.com/charliermarsh/ruff/blob/main/src/rules/flake8_annotations/rules.rs#L26

Reply:
I think we could add `ClassDef` there. However it's probably just an optimization in practice... If a `return` is inside a `ClassDef`, but _not_ inside a `method`, you get a syntax error anyway.

Reply:
yea makes sense